docs/references/nextjs/clerk-middleware.mdx

```diff
@@ -0,0 +1,239 @@
+---
+title: clerkMiddleware() | Next.js
+description: The `clerkMiddleware()` function allows you to protect your Next.js application using middleware.
+---
+
+# `clerkMiddleware()`
+
+The `clerkMiddleware()` helper integrates Clerk authentication into your Next.js application through middleware. `clerkMiddleware()` is compatible with both the App and Pages routers. 
+
+## Usage
+
+Create a `middleware.ts` file. If your application uses the `src/` directory, your `middleware.ts` file should be placed inside the `src/` folder. If you are not using a `src/` directory, place the `middleware.ts` file in your root directory. 
+
+<Callout level="info">
+For more information about middleware in Next.js, see the [Next.js documentation](https://nextjs.org/docs/pages/building-your-application/routing/middleware).
+</Callout>
+
+```tsx filename="middleware.ts"
+import { clerkMiddleware } from '@clerk/nextjs/server';
+
+export default clerkMiddleware()
+
+export const config = {
+  matcher: ['/((?!.*\\..*|_next).*)', '/', '/(api|trpc)(.*)'],
+};
+````
+
+By default `clerkMiddleware` will not protect any routes. With `clerkMiddleware()` all routes are now public and you must opt-in to protection for routes.
+
+With the recommended matcher, all routes are matched by Clerk's authentication Middleware, with the exception of internal `/_next/` routes and static files. Static files are detected by matching on paths that end in `.+\..+`.
+
+## Environment Variables
+
+If you are building your own sign-in pages, you don't need to add these pages to your `publicRoutes`. They will be inferred from your `.env` file:
+
+```ts filename=".env"
+NEXT_PUBLIC_CLERK_SIGN_IN_URL=/sign-in
+NEXT_PUBLIC_CLERK_SIGN_UP_URL=/sign-up
+NEXT_PUBLIC_CLERK_AFTER_SIGN_IN_URL=/
+NEXT_PUBLIC_CLERK_AFTER_SIGN_UP_URL=/
+```
+
+## Protect Multiple Routes
+
+`clerkMiddleare()` ships with the helper function `createRouteMatcher()`. `createRouteMatcher()` allows you to add an array of routes, which returns a function that takes in the `request` object to then check if the route the user is trying to visit matches one of the routes. You can use this matcher to protect multiple routes. The paths provided to this helper can be in the same format as the paths provided to the Next Middleware matcher.  
+
+In the following example the routes `/dashboard(.*)` and `/forum(.*)` are passed to the `createRouteMatcher()` function and assigned to `isProtectedRoute`.
+
+
+```tsx filename="middleware.ts"
+import { clerkMiddleware, createRouteMatcher } from '@clerk/nextjs/server';
```

Comment:
Reduce line length here to not have horizontal scroll bars on the code component

Reply:
Is there a way we can do this programmatically? Either prevent horizontal scroll or a linter that detects it?